facebook/xcbuild
# [process] Default context creation now uses main's argv and argc directly

Instead of relying on a constructor to grab the argc/argv (which doesn't
work on my Centos7 machine), use the argc/argv passed in from main().

This also defines the global default context to whatever is created in
each executable's main() function instead of creating a new object.

Test plan:
- run 'plutil' on my Linux machine (after adding "-lpthread" added to the build invocation)
- run 'plutil' on my OSX machine

See https://github.com/facebook/xcbuild/issues/138
